db/migrate/20241023082630_create_enrollments.rb

```diff
@@ -0,0 +1,15 @@
+class CreateEnrollments < ActiveRecord::Migration[7.2]
+  def up
+    create_table :enrollments do |t|
+      t.integer :user_id
```

Comment:
Can you check if this works?
```suggestion
      t.references :user, null: false, foreign_key: true
```

Reply:
Lets verify this for other migrations too, and add the necessary null checks